source/extensions/filters/network/rbac/rbac_filter.cc

```diff
@@ -13,14 +14,47 @@ namespace Extensions {
 namespace NetworkFilters {
 namespace RBACFilter {
 
+absl::Status ActionValidationVisitor::performDataInputValidation(
+    const Envoy::Matcher::DataInputFactory<Filters::Common::RBAC::Matching::MatchingData>&,
+    absl::string_view type_url) {
+  static absl::flat_hash_set<std::string> allowed_inputs_set{
+      {TypeUtil::descriptorFullNameToTypeUrl(
+          envoy::extensions::matching::common_inputs::network::v3::DestinationIPInput::descriptor()
+              ->full_name())},
+      {TypeUtil::descriptorFullNameToTypeUrl(envoy::extensions::matching::common_inputs::network::
+                                                 v3::DestinationPortInput::descriptor()
+                                                     ->full_name())},
+      {TypeUtil::descriptorFullNameToTypeUrl(
+          envoy::extensions::matching::common_inputs::network::v3::SourceIPInput::descriptor()
+              ->full_name())},
+      {TypeUtil::descriptorFullNameToTypeUrl(
+          envoy::extensions::matching::common_inputs::network::v3::SourcePortInput::descriptor()
+              ->full_name())},
+      {TypeUtil::descriptorFullNameToTypeUrl(
+          envoy::extensions::matching::common_inputs::network::v3::DirectSourceIPInput::descriptor()
+              ->full_name())},
+      {TypeUtil::descriptorFullNameToTypeUrl(
+          envoy::extensions::matching::common_inputs::network::v3::ServerNameInput::descriptor()
+              ->full_name())}};
+  if (allowed_inputs_set.contains(type_url)) {
+    return absl::OkStatus();
+  }
+
+  return absl::InvalidArgumentError(
+      fmt::format("RBAC network filter cannot match on HTTP inputs, saw {}", type_url));
```

Comment:
Can we also include the supported list of matchers (defined above) in the output message? Also do you mean all the other matchers (not in the list above) are HTTP matchers? (I'm not sure if this will be always the case in the future)

Reply:
In fact, we have documented supported inputs [here](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/matching/matching_api#inputs-and-matching-algorithms). RBAC network and HTTP filter share the same engine, and the matching data is registered in engine. So we have to exclude unsupported HTTP inputs for RBAC network filter, which is the same behavior of `validateRbacRules`.

Reply:
Done.